ion-schema/src/test/kotlin/com/amazon/ionschema/TypeTest.kt

```diff
@@ -64,7 +64,6 @@ class TypeTest {
         val iter = violations.iterator()
         val violation = iter.next()
         assertEquals("type_mismatch", violation.code)
-        assertEquals("type", violation.constraint?.fieldName)
```

Comment:
(Question) Why was this removed? I couldn't find any changes in `Violation` structure for this PR.

Reply:
The old way of creating the spec-defined types involved reading an Ion struct and then extracting the symbol from the struct. This meant that the symbols had the field name "type". See https://github.com/amazon-ion/ion-schema-kotlin/blob/a3fabe9a472095b23a2aa16c810da3f5b26734c4/ion-schema/src/main/kotlin/com/amazon/ionschema/internal/SchemaCore.kt#L86-L118

Now we create the symbol directly instead, which means that the symbol no longer has a field name. (This whole thing is a weird consequence of the unfortunate decision that an `IonValue` knows its own field name.) Anyway, this lack of field name causes no difference in the violation messages, and the violation constraint field name is not actually used anywhere in the library.